### PR TITLE
dialects: Initial math dialect

### PR DESCRIPTION
**Pull request by mesham**
Initial math dialect for sqrt and pow operations, checked into experimental as no tests but needed for benchmarks - other operations will likely be needed to but will commit in stages as they are required.

### CODE REVIEW

**Review by georgebisbas — COMMENTED** [review]
Just discussing about this, and was your PR, nice timing

File: xdsl/dialects/experimental/math.py
Comment (webmiche) [inline xdsl/dialects/experimental/math.py:48]: Why not for rhs as well?

**Review by webmiche — APPROVED** [review]
LGTM.

File: xdsl/dialects/experimental/math.py
Comment (mesham) [reply xdsl/dialects/experimental/math.py:48]: I just did this for the typing to keep PyRight happy as we grab out the .type, I don't think we really need it apart from keeping PyRight happy? But happy to do for RHS too if that keeps it all the same?

Comment (webmiche) [reply xdsl/dialects/experimental/math.py:48]: Ah I see, yeah they differ there. I think, this code is bound to break at some point in the future, when we make build more restrictive, but we can also handle the issue then, I don't mind.

**Review by AntonLydike — APPROVED** [review]
LGTM, if you want I can clean up the last pyright errors before we merge as well!

File: xdsl/dialects/experimental/math.py
Comment (mesham) [reply xdsl/dialects/experimental/math.py:48]: I've updated it to do this for RHS too so that they are the same

**Pull request merged**
